feat: Update Postgres search docs

dshukertjr opened the pull request:
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

- Update the YouTube video
- Add deeper guidelines on text search includging
  - Using websearch for generating tsquery
  - Ranking the search results by relevance
  - Adding weights to different columns

dshukertjr committed:
feat: Update the video to the latest text search video

dshukertjr committed:
remove the embedded video for text search, and update the tocVideo

dshukertjr committed:
Add a section to dig deeper into webseach function

dshukertjr committed:
Add section about ranking

dshukertjr committed:
Add some weight sample

dshukertjr committed:
run formatter

dshukertjr committed:
Fix reviewdog issues

dshukertjr committed:
missed a spot

dshukertjr committed:
Update the spelling rules

dshukertjr committed:
merge main

charislam committed:
feat(cron jobs): manual run tooltip (#39600)

Add a tooltip explaining that manual runs don't show up as cron jobs.
Some users were confused about this.

dnywh committed:
studio(chore): improve analytics bucket page (#39545)

* specialized dialog for analytics (and vector) bucket creation

* handle missing wrapper extension and iceberg wrapper integration

* bucket creation improvements

* extension and integration checks

* dialog header consistency

* simplify bucket dialog creation button trigger

* fix subtitle ternary

* example bucket without dynamic route

* create dynamic segment

* fancy connection table

* improve bucket page

* connection table polish

* polish

* delete bucket modal

* improve missing installations design

* new table modal

* fix copy icon

* copy all env button placeholder

* list all analytics buckets

* specialised bucket modal

* shared installer hooks and components

* make iceberg wrapper not dependent on wrapper extension

There are cases where the Iceberg Wrapper can be installed but the Wrapper extension has since been removed

* better admonition copywriting

* todo

* misc

* simplify

* simplifications

* work within existing page

* improvements to existing route

* v2 tables

* simplify

* new table modal

* change route to thin wrapper

* reduce size of PR

* remove double scaffold element

* minor nits

---------

Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

SaxonF committed:
AI model selection (#39472)

* expose model and model selection

* tooltip

* simplify

* Update ai-assistant-state.tsx

Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

* Update ModelSelector.tsx

---------

Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

sweatybridge committed:
fix: update github app name for nimbus (#39606)

alaister committed:
fix: plan upgrade url for org usage pages (#39608)

* fix: plan upgrade url for org usage pages

* nit fix

* fix

---------

Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

ivasilov committed:
feat: Redirect the user after claiming a project (#39527)

* Use the url in the response to redirect the user.

* Fix a title warning.

* Update apps/studio/pages/claim-project.tsx

Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

---------

Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

fsansalvadore committed:
change connection string via dropdown (#39354)

* change connection string via dropdown
* rearrange descriptive content and actions in connection string tab
* update connection string url state based on selected type and method
* reset query states when switching tabs
* fix steps layout in Connection String dialog tab
* rounded-lg on mobile
* update wording of step 3 on connection string tab
* improve map key naming

---------

Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

alaister committed:
[FE-1938] feat: show region capacity issues (#39611)

feat: show region capacity issues

charislam committed:
fix(users search): add explanation for 'all columns' search (#39599)

* fix(users search): add explanation for 'all columns' search

* Tiny

---------

Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

ChrisChinchilla committed:
docs: Remove unused scripts (#39595)

Remove unused scripts

ignaciodob committed:
feat: entitlements (#39338)

* feat: entitlements

* Sample usage

* format

* Reworked

* format

* gen

* removed

* removed

ChrisChinchilla committed:
docs: Add configurable padding to docs header logo (#39615)

* Add configurable padding to docs header logo

* ci: Autofix updates from GitHub workflow

* tweak

---------

Co-authored-by: alaister <10985857+alaister@users.noreply.github.com>
Co-authored-by: Alaister Young <a@alaisteryoung.com>

fsansalvadore committed:
remove connection string badge in dropdown (#39617)

charislam committed:
feat(preferences): allow disable hotkeys (#39573)

* feat(preferences): allow disable hotkeys

Add a section in /account/me for disabling hotkeys. Only added one
hotkey for now (Cmd + E for toggling editor side panel) but we can add
more with the same pattern.

* refactor: remove default export on ProjectLayout

* feat(hotkeys): allow toggling of command menu and ai assistant hotkeys

* Nit

* PRettier lint

---------

Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

jordienr committed:
FE-1955 fix query perf links (#39622)

fix

github-actions[bot] committed:
docs: update js client libraries (v2.75.1) (#39620)

Co-authored-by: supabase-releaser[bot] <223506987+supabase-releaser[bot]@users.noreply.github.com>
Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

github-actions[bot] committed:
feat: update @supabase/*-js libraries to v2.75.1 (#39619)

Co-authored-by: supabase-releaser[bot] <223506987+supabase-releaser[bot]@users.noreply.github.com>

joshenlim committed:
Fix project card instance size missing (#39616)

* Fix project card instance size missing

* Fix types

joshenlim committed:
Nit tidy up hotkey settings (#39628)

ivasilov committed:
Update the command to use the @supabase alias (#39627)

Update the command to use the @supabase alias.

joshenlim committed:
Chore/persist auth users search configuration (#39649)

* Perist auth users search configuration

* nit

* Clean up

* Swap more params to use URL state

* Clean up

ChrisChinchilla commented on apps/docs/content/guides/database/full-text-search.mdx:
```diff
@@ -751,6 +749,81 @@ When you want the search term to include a phrase or multiple words, you can con
+
 ## Creating indexes
 
 Now that we have Full Text Search working, let's create an `index`. This will allow Postgres to "build" the documents preemptively so that they
```
```suggestion
Now that you have Full Text Search working, create an `index`. This allows Postgres to "build" the documents preemptively so that they
```

Nothing to do with you, but while we're here let's fix some things :)

ChrisChinchilla commented on apps/docs/content/guides/database/full-text-search.mdx:
```diff
@@ -1142,6 +1215,323 @@ data = client.from_('books').select().text_search('description', "'big' & !'litt
 
+## Ranking search results [#ranking]
+
+Postgres provides ranking functions to sort search results by relevance, helping you present the most relevant matches first. Since ranking functions need to be computed server-side, we'll use RPC functions and generated columns.
```
```suggestion
Postgres provides ranking functions to sort search results by relevance, helping you present the most relevant matches first. Since ranking functions need to be computed server-side, use RPC functions and generated columns.
```

ChrisChinchilla reviewed (changes requested):
@dshukertjr Excellent stuff! Mostly just some small grammar tweaks…

sweatybridge committed:
fix: allow merge request with stale PR to be closed (#39670)

* fix: allow merge request with stale PR to be closed

* chore: comment on review requested field

charislam committed:
fix: increase retry delay for throttled api requests (#39634)

Increasing this because of reports that CSV uploads are hitting 429s. (I
checked and there are no Retry-After headers, so we should be falling
back to the default exponential backoff.)

Also brings the implementation in line with the comment :P Before, the
comment claimed we were starting at 1s but we were starting at 500ms.

charislam committed:
chore: update github workflows (#39237)

aaronByrne1 committed:
troubleshooting guide for users seeing empty branches due to no migrations (#39293)

* New troubleshooting guide for users seeing empty branches due to no migration files

* format

* format and changes

* format

* Change title

* Addressed initial comments

theccres committed:
Add Chase Cresgy to humans.txt (#39675)

jordienr committed:
fix db connections free (#39624)

fix

mattrossman committed:
feat: self-hosted MCP debugging tools (#39366)

* feat: stub `getDebuggingOperations`

* fix: logflare variables in `generateLocalEnv.js`

* chore: upgrade `supabase`, `@supabase/mcp-server-supabase`

* feat: shared `retrieveAnalyticsData`, implement `getLogs` for MCP

* feat: shared `getLints`, implement MCP security/performance advisors with filter

* chore: update lockfile

* fix: prefer `??`

Co-authored-by: Greg Richardson <greg.nmr@gmail.com>

* docs: comment origin of `LINT_SQL`

Co-authored-by: Greg Richardson <greg.nmr@gmail.com>

* chore: bump `supabase` devDependency for latest images

Changes rely on fix from loglare v1.23.0

* fix: edge function logs query

Based on https://github.com/supabase/supabase/pull/39388/files#diff-b4e9726d34a406fa1a6133a18bdb972c90be9df0c89598de0695592514e22941R199-R206

---------

Co-authored-by: Greg Richardson <greg.nmr@gmail.com>

charislam committed:
fix: add time zone to commit time and add to support message (#39629)

* fix: add time zone to commit time and add to support message

The time zone isn't getting logged on the commit times, which makes it
confusing what the actual time is. Also added it to the support message
alongside the version number, to help quickly rule out outdated version
issues.

* Nit unrelated update auth users sort dropdown copy

---------

Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

dshukertjr committed:
Update apps/docs/content/guides/database/full-text-search.mdx

Co-authored-by: Chris Chinchilla <chris.ward@supabase.io>

dshukertjr committed:
Update apps/docs/content/guides/database/full-text-search.mdx

Co-authored-by: Chris Chinchilla <chris.ward@supabase.io>

dshukertjr committed:
Update apps/docs/content/guides/database/full-text-search.mdx

Co-authored-by: Chris Chinchilla <chris.ward@supabase.io>

dshukertjr committed:
Update apps/docs/content/guides/database/full-text-search.mdx

Co-authored-by: Chris Chinchilla <chris.ward@supabase.io>

dshukertjr committed:
Update apps/docs/content/guides/database/full-text-search.mdx

Co-authored-by: Chris Chinchilla <chris.ward@supabase.io>

dshukertjr committed:
Update apps/docs/content/guides/database/full-text-search.mdx

Co-authored-by: Chris Chinchilla <chris.ward@supabase.io>

dnywh committed:
chore(studio): add feature preview for auth security template UI (#39661)

* feature preview logic

* sidebar changes based on feature preview

* singular naming

* minor

* Flip isPlatformOnly for new preview

---------

Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

joshenlim committed:
Chore/simplified support form (#39554)

* Simplified support form

* Nit

kevcodez committed:
docs: TP MAU and Database size quotas (#39699)

Quotas have been relaxed a long time ago, missed this in the docs

kevcodez committed:
docs: edge function preflight usage docs (#39700)

* docs: edge function preflight usage docs

* Update edge-function-invocations.mdx

kiwicopple committed:
chore: update self-hosted image versions (#39468)

* chore: update image versions for docker/docker-compose.yml

* chore: update image versions for docker/docker-compose.yml

* Update docker-compose.yml

---------

Co-authored-by: Han Qiao <sweatybridge@gmail.com>

ChrisChinchilla reviewed (approved):
🚀

joshenlim committed:
Silence text model error in diff editor (#39703)

sweatybridge committed:
fix: cron logs query to handle type cast (#39704)

sweatybridge committed:
fix: use separate query clause for self hosted (#39711)

hf committed:
feat: add navigator lock debugging to track down some rare issue (#39688)

* feat: add navigator lock debugging to track down some rare issue

* send exception to sentry

cemalkilic committed:
feat(auth): use fetch with timeout for auth calls (#39714)

filipecabaco committed:
fix: improve realtime inspector (#39626)

* Add better error descriptions when the channel fails to join
* If there are no publications, database changes will be disabled and untoggled to prevent errors

CoolAssPuppy committed:
Added new case study for Rally (#39721)

* Added new case study for Rally

* ci: Autofix updates from GitHub workflow

* chore: colour correct rally logos

---------

Co-authored-by: kemal <hello@kemal.earth>

charislam committed:
feat(region selector): get region status by desired size (#39631)

* feat(region selector): get region status by desired size

The available-regions endpoint now takes a desired_instance_size query
parameter to return more accurate status data.

* Align call of useOrganizationAvailableRegionsQuery in [slug] with RegionSelector, and rearrange some consts

---------

Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

hf committed:
feat: bump debug time of auth lock to 10s (#39724)

awaseem committed:
Updated docs to use pnpm for studio (#39727)

updated docs to use pnpm

stylessh committed:
update contributors count at /careers page (#39598)

upd count

wong2 committed:
docs: fix apple secret generator tool form spacing (#39612)

Co-authored-by: Alan Daniel <stylesshjs@gmail.com>

hf committed:
feat: reduce profile information requests, faster page renders (#39729)

awaseem committed:
Add Ali Waseem to Humans.txt (#39726)

added myself to human.txt

charislam committed:
ci(dependabot): don't get prompted for major version updates for github actions (#39733)

Major version updates require more preparation and testing, so we only
want to update these on our schedule / if actually deprecated, not be
prompted as soon as a new major version comes out. We mostly want to
keep patch/minor updates so we can be aware of any vulnerability fixes.

charislam committed:
docs(readme): fix dev instructions for internal/external (#39728)

charislam committed:
cleanup(region selector): remove code from selected region (#39730)

* cleanup(region selector): remove code from selected region

Selector looks a bit busy and you probably know the code if you've
already selected it, so removing it from the dropdown as suggested in a
previous PR.

* refactor(region selector): clean up unnecessary type checks

Co-authored-by: Ali Waseem <waseema393@gmail.com>

* ci: Autofix updates from GitHub workflow

---------

Co-authored-by: Ali Waseem <waseema393@gmail.com>

kanishkdudeja committed:
fix(billing): make balance optional in subscription endpoint response (#39746)

saltcod committed:
Swap project query to fix team role picker (#39720)

* Swap project query to fix team role picker

* Cleanup

* Fetch the projects in all cases, instead of only when there's no scoped roles.

* Make the projects fetch only when the panel is visible.

* Nit

---------

Co-authored-by: Ivan Vasilov <vasilov.ivan@gmail.com>
Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

charislam committed:
fix(policy editor): reduce number of table role access queries (#39601)

* fix(policy editor): reduce number of table role access queries

On the Policies page, every single PolicyTableRow fires off its own
table roles access query (because each query is keyed to an individual
table). For users with many tables (400+), this causes the API to
rate-limit the dashboard.

Changed so that the table roles access query now fetches information for
the entire schema in one go, then organizes it by table.

* Nit refactors

* refactor: return minimal data from table access query

All we need to know is whether a given table has anon or authenticated
access, so just query for the list of tables that match that criteria
from information_schema.role_table_grants.

* refactor(policies page): more efficient sql query for getting anon/auth access tables

---------

Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

SaxonF committed:
Fix gap issue (#39754)

fix gap issue

joshenlim committed:
Use upgrade plan button in FreeProjectLimitWarning (#39715)

dshukertjr committed:
run formatter
